src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,227 @@
+        }
+
+        /// <summary>
+        /// Get current authorized Oracle validator
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Authorized Oracle validator</returns>
+        public ECPoint[] GetOracleValidators(StoreView snapshot)
+        {
+            ECPoint[] cnPubKeys = NativeContract.NEO.GetValidators(snapshot);
+            ECPoint[] oraclePubKeys = new ECPoint[cnPubKeys.Length];
+            System.Array.Copy(cnPubKeys, oraclePubKeys, cnPubKeys.Length);
+            for (int index = 0; index < oraclePubKeys.Length; index++)
+            {
+                var oraclePubKey = oraclePubKeys[index];
+                StorageKey key = CreateStorageKey(Prefix_Validator, oraclePubKey);
+                ECPoint delegatePubKey = snapshot.Storages.TryGet(key)?.Value.AsSerializable<ECPoint>();
+                if (delegatePubKey != null) { oraclePubKeys[index] = delegatePubKey; }
+            }
+            return oraclePubKeys;
+        }
+
+        /// <summary>
+        /// Get number of current authorized Oracle validator
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>The number of authorized Oracle validator</returns>
+        [ContractMethod(0_01000000, ContractParameterType.Integer)]
+        private StackItem GetOracleValidatorsCount(ApplicationEngine engine, Array args)
+        {
+            return GetOracleValidatorsCount(engine.Snapshot);
+        }
+
+        /// <summary>
+        /// Get number of current authorized Oracle validator
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>The number of authorized Oracle validator</returns>
+        public BigInteger GetOracleValidatorsCount(StoreView snapshot)
+        {
+            return GetOracleValidators(snapshot).Length;
+        }
+
+        /// <summary>
+        /// Create a Oracle multisignature address
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Oracle multisignature address</returns>
+        public UInt160 GetOracleMultiSigAddress(StoreView snapshot)
+        {
+            ECPoint[] cnPubKeys = NativeContract.NEO.GetValidators(snapshot);
+            return Contract.CreateMultiSigRedeemScript(cnPubKeys.Length - (cnPubKeys.Length - 1) / 3, cnPubKeys).ToScriptHash();
+        }
+
+        /// <summary>
+        /// Set HttpConfig
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Returns true if the execution is successful, otherwise returns false</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.Integer }, ParameterNames = new[] { "httpConfig" })]
+        private StackItem SetHttpConfig(ApplicationEngine engine, Array args)
```

Comment:
If we have multiple settings in `HttpConfig`, do we have to set all of them at once?

Reply:
Yes, but if you have other ideas, I can change

Reply:
Maybe we can get or set it by key. For example:

```c#
GetConfig("HttpTimeout");
```

```c#
SetConfig("HttpTimeout", 2000);
```

Reply:
@shargon What's your opinion?

Reply:
Use this way?
```js
SetHttpConfig("Timeout", 2000) // put: http+timeout => 2000
GetHttpConfig()  // return all the http configurations
```

Reply:
If we use `enum` or `const` for the name, it's good for me :)